pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -488,14 +477,21 @@ impl<T: Config> Module<T> {
                     );
                 }
             }
+
+            // If the contract is in canceled by user state, set the contract in kill state (end state)
+            if contract.state == types::ContractState::Deleted(types::Cause::CanceledByUser) {
+                Self::kill_contract(&mut contract);
+            } else {
+                // prepare the contract to be billed at the next billing cycle
+                Self::_reinsert_contract_to_bill(contract.contract_id);
+            }
```

Comment:
The if and else here do 2 separate things: the if block does a state transition, whereas the else controls logic to reinsert in the billing cycle. Personally I'd prefer this to be split into 2 blocks:

```
// If the contract is in canceled by user state, set the contract in kill state (end state)
if matches!(contract.state, types::ContractState::Deleted(types::Cause::CanceledByUserr) {
    Self::kill_contract(&mut contract);
}

// reinsert contracts in billing loop if they are not in terminal state
if !matches(contract.state, types::ContractState::Killed) {
    // prepare the contract to be billed at the next billing cycle
    Self::_reinsert_contract_to_bill(contract.contract_id);
}
```